src/main/java/org/opensearch/sdk/BaseExtension.java

```diff
@@ -35,12 +36,37 @@ public abstract class BaseExtension implements Extension {
     protected ThreadPool threadPool;
 
     /**
-     * Empty constructor to fulfill abstract class requirements
+     * The extension settings include a name, host address, and port.
+     */
+    private ExtensionSettings settings;
+
+    /**
+     * Empty constructor to fulfill abastract class requirement
      */
     protected BaseExtension() {
 
     }
 
+    /**
+     * Instantiate this extension, initializing the connection settings and REST actions.
+     * @throws IOException on failure to load settings.
+     */
+    protected BaseExtension(String path) throws IOException {
```

Comment:
This does not throw `IOException` as it is caught, and an `ExceptionInInitializerError` is thrown instead.  That doesn't need to be added to a throws block as it's a runtime exception.